kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaProtocolHandler.java

```diff
@@ -464,4 +459,66 @@ private void loadTxnLogTopics(TransactionCoordinator txnCoordinator) throws Exce
             log.info("Current broker: {} does not own any of the txn log topic partitions", currentBroker);
         }
     }
+
+    private static CompletableFuture<InetSocketAddress> getFailedAddressFuture(final Throwable throwable) {
+        final CompletableFuture<InetSocketAddress> future = new CompletableFuture<>();
+        future.completeExceptionally(throwable);
+        return future;
+    }
+
+    private static CompletableFuture<InetSocketAddress> getAddressFutureFromBrokerUrl(final String brokerUrl) {
+        final CompletableFuture<InetSocketAddress> future = new CompletableFuture<>();
+        try {
+            final URI uri = new URI(brokerUrl);
+            future.complete(InetSocketAddress.createUnresolved(uri.getHost(), uri.getPort()));
+        } catch (URISyntaxException e) {
+            future.completeExceptionally(e);
+        }
+        return future;
+    }
+
+    public static CompletableFuture<InetSocketAddress> getBroker(final PulsarService pulsarService,
+                                                                 final TopicName topicName) {
+        final NamespaceService namespaceService = pulsarService.getNamespaceService();
+        if (namespaceService == null) {
+            return getFailedAddressFuture(new IllegalStateException("NamespaceService is not available"));
+        }
+
+        final LookupOptions options = LookupOptions.builder()
+                .authoritative(false)
+                .advertisedListenerName(null)
```

Comment:
@wangjialing218 I think we can set advertised listener name here like https://github.com/streamnative/kop/issues/644#issuecomment-895770747. But it's not clear that how to get the listener name?

Reply:
we could add configration in broker.conf like this in k8s:

> kafkaListenerName=kafkaListener
> kafkaListeners=PLAINTEXT://0.0.0.0:9092
> kafkaAdvertisedListeners=PLAINTEXT://{k8s node ip}:{nodeport of 9092}
> advertisedListeners=internal:pulsar://{pod clusterIP}:6650,external:pulsar://{k8s node ip}:{nodeport of 6650},kafkaListener:pulsar://{k8s node ip}:{nodeport of 9092}

and we set advertised listener here with the value of `kafkaListenerName`

We could get the lookup result for kafka advertised address `{k8s node ip}:{nodeport of 9092}` by set listener name. No neet to convert the broker address to kafka advertised address through `KopBrokerLookupManager.getProtocolDataToAdvertise`, which may cause the issue in #633

Reply:
If I understand correctly, set `advertisedListenerName` to the configured `kafkaListenerName`, right?

> No neet to convert the broker address to kafka advertised address through KopBrokerLookupManager.getProtocolDataToAdvertise

Regarding to this problem, I'll read the related code and confirm it soon.

Reply:
> If I understand correctly, set `advertisedListenerName` to the configured `kafkaListenerName`, right?
Yes. Current we support only one kafka advertised listener, we could set the listener name in configuration.
But kafka support multi advertised listeners, when we start the work for #574, we may need to consider how to handle multi advertised listener case.